xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsFormatterProvider.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.mentions.internal;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.component.manager.ComponentManager;
+import org.xwiki.mentions.MentionsFormatter;
+
+import static org.apache.commons.lang3.exception.ExceptionUtils.getRootCauseMessage;
+
+/**
+ * Default implementation of {@link MentionFormatterProvider}.
+ * <p>
+ * A lookup of the requested formatter type is performed.
+ * An anonymous instance of {@link MentionsFormatter} is returned if the lookup fails.
+ *
+ * @version $Id$
+ * @since 12.10RC1
+ */
+@Component
+@Singleton
+public class DefaultMentionsFormatterProvider implements MentionFormatterProvider
+{
+    @Inject
+    private ComponentManager componentManager;
+
+    @Inject
+    private Logger logger;
+
+    /**
+     * Anonymous instance of the mention formater. Used as a default implementation if no specific implementation is
+     * found for a given type.
+     */
+    private final MentionsFormatter defaultFormatter = new MentionsFormatter()
```

Comment:
I'm not a big fan, why not having a `DefaultMentionsFormatter` component that you'd inject here? It sounds safer to future extensions of this work.

Reply:
`DefaultMentionsFormatter` (re-)introduced.